text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,1777 @@
+              knownMessages: 0
+              receivedMessageRequests: 0
+              receivedMilestoneRequests: 0
+              receivedHeartbeats: 0
+              sentMessages: 0
+              sentMessageRequests: 0
+              sentMilestoneRequests: 0
+              sentHeartbeats: 0
+              droppedPackets: 0
+  schemas:
+
+    Message:
+      description: A message is the object nodes gossip around in the network. It always references two other messages that are known as parents. It is stored as a vertex on the tangle data structure that the nodes maintain.
+      properties:
+        networkId:
+          type: string
+          description: Network identifier. This field signifies for which network the message is meant for. It also tells which protocol rules apply to the message. It is computed out of the first 8 bytes of the `BLAKE2b-256` hash of the concatenation of the network type and protocol version string.
+        parent1MessageId:
+          type: string
+          description: The identifier of the first message this message references.
+        parent2MessageId:
+          type: string
+          description: The identifier of the second message this message references.
+        payload:
+          oneOf:
+            - $ref: '#/components/schemas/TransactionPayload'
+            - $ref: '#/components/schemas/MilestonePayload'
+            - $ref: '#/components/schemas/IndexationPayload'
+        nonce:
+          type: string
+          description: The nonce which lets this message fulfill the Proof-of-Work requirement.
+
+    TransactionPayload:
+      description: The Transaction Payload to be embedded into a message.
+      properties:
+        type:
+          type: integer
+          description: Set to value 0 to denote a Transaction Payload.
+        essence:
+          type: object
+          oneOf:
+            - $ref: '#/components/schemas/TransactionEssence'
+        unlockBlocks:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/SignatureUnlockBlock'
+              - $ref: '#/components/schemas/ReferenceUnlockBlock'
+
+    TransactionEssence:
+      description: Describes the essence data making up a transaction by defining its inputs and outputs and an optional payload.
+      properties:
+        type:
+          type: integer
+          description: Set to value 0 to denote a Transaction Essence.
+        inputs:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/UTXOInput'
+        outputs:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/SigLockedSingleOutput'
+        payload:
+          oneOf:
+            - $ref: '#/components/schemas/IndexationPayload'
+          nullable: true
+
+    UTXOInput:
+      description: Describes an input which references an unspent transaction output to consume.
+      properties:
```

Comment:
Needs required field e.g. `required: ["type", "transactionId", "transactionOutputIndex"]` 

This applies to pretty much all the objects, by default everything is marked as optional.